### PR TITLE
OpenSSL only if feature networking is set.

### PR DESCRIPTION
**Pull request by dok-net**
This fixes #141 to not include OpenSSL if there's no networking selected.

### CODE REVIEW

**Comment by dok-net** [comment]
This alone doesn't turn off bulding the tests in the submodule expected, which takes a very long time. Any ideas?

**Comment by jbaldwin** [comment]
Ah yeah I missed this, thanks for opening a PR.  I'll need to think on the tests, but maybe that part can just be disabled as well, I think it's just making a unique SSL cert per run.

**Comment by dok-net** [comment]
@jbaldwin I've added the actual cmake line I was using, I've disabled tests anyway. See how the `expected` lib builds their tests, anyway. What about C++23 previews, can the expected submodule get ignored for those already, and if, how?

**Comment by jbaldwin** [comment]
Can you open an issue for the expected lib building tests when tests are disabled?  Its a bug in the build system for sure, it might affect the other vendor deps as well

**Comment by jbaldwin** [comment]
We can use the expected vendor for c++20 and for c++23 use the stdlib version, its possible via cmake to detect the c++ version and only include the vendor one if its < c++23 basically.  I would make that as a different issue as well.  Realistically I think targeting c++20 for this library indefinitely makes the most sense though, but lets pickup new features if the user is using a new version!

**Comment by jbaldwin** [comment]
I'm not exactly sure I can add a commit to your fork's PR so I just opened another PR here with the additions to make the tests work properly:

https://github.com/jbaldwin/libcoro/pull/144

**Comment by dok-net** [comment]
The cmake configure step still probes for all the networking related APIs:

**Comment by dok-net** [comment]
Can the c-ares be excluded, depending on the networking feature selection?

**Comment by dok-net** [comment]
> I'm not exactly sure I can add a commit to your fork's PR so I just opened another PR here with the additions to make the tests work properly:
> 
> #144

I've included your changes into my PR branch.

**Comment by dok-net** [comment]
Lots of "catch-2" build steps still occur, though.

**Comment by jbaldwin** [comment]
Yes, I think that should be fixed on the other PR here https://github.com/jbaldwin/libcoro/issues/143, this is just removing openssl with `LIBCORO_FEATURE_NETWORKING=OFF`.  Trying to keep them distinct commits since they are different issues.

**Comment by dok-net** [comment]
I'm not merging the pending PRs and pushing, just have it all at once in my work tree and it's not completely there, just yet.

**Comment by dok-net** [comment]
@jbaldwin This PR includes your branch `origin/openssl_only_feature_networking` completely, and finishes the remaining issue with "c-ares". Please review, it's ready to merge from my side.

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,23 +31,26 @@ message("${PROJECT_NAME} LIBCORO_BUILD_EXAMPLES        = ${LIBCORO_BUILD_EXAMPLE
...
-
-    add_subdirectory(vendor/c-ares/c-ares)
     add_subdirectory(vendor/tartanllama/expected)
 else()
-    find_package(c-ares CONFIG REQUIRED)
     find_package(tl-expected CONFIG REQUIRED)
 endif()
 
-find_package(OpenSSL REQUIRED)
+if(LIBCORO_FEATURE_NETWORKING)
```
[the remark this answers is not in the document]
Comment (jbaldwin) [reply CMakeLists.txt:43]: I also realize around line 83 its re-checked again, moving source file declaration up might reduce redundant checks even more, but I can probably change that later if you don't want to deal with it.

File: CMakeLists.txt [hunk 2]
```diff
@@ -98,8 +102,10 @@ add_library(${PROJECT_NAME} STATIC ${LIBCORO_SOURCE_FILES})
 set_target_properties(${PROJECT_NAME} PROPERTIES LINKER_LANGUAGE CXX PREFIX "")
 target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_20)
 target_include_directories(${PROJECT_NAME} PUBLIC inc)
-target_link_libraries(${PROJECT_NAME} PUBLIC pthread c-ares::cares tl::expected OpenSSL::SSL OpenSSL::Crypto)
+target_link_libraries(${PROJECT_NAME} PUBLIC pthread tl::expected)
 if(LIBCORO_FEATURE_NETWORKING)
+    target_link_libraries(${PROJECT_NAME} PUBLIC c-ares::cares)
```
[the remark this answers is not in the document]
Comment (dok-net) [reply CMakeLists.txt:107]: Done.

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (dok-net) [reply CMakeLists.txt:43]: Part one, done, part two, I think that causes duplication?

**Review by jbaldwin — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (jbaldwin) [reply CMakeLists.txt]: Looks better and a lot easier to follow IMO, thanks for changing it.

**Pull request merged**